tests/test_middlewares.py

```diff
@@ -13,15 +16,45 @@ async def get_application(self):
         """Retrieve web Application for test."""
         return await init()
 
-    @unittest_run_loop
-    async def test_bad_HTTP_request_converts_into_json_response(self):
-        """Test that middleware reformats 400 error with problem details."""
+    async def setUpAsync(self):
+        """Configure default values for testing JWTs.
+
+        Also set pem as an environment variable for the duration of the tests.
+        """
+        self.header = {'alg': "HS256", 'typ': "JWT"}
+        self.payload = {
+            'sub': "test",
+            'name': "tester",
+            'iss': "haka_iss",
+            'exp': 9999999999
+        }
+        self.pem = {
+            'kty': "oct",
+            'alg': "RS256",
+            'k': "GawgguFyGrWKav7AX4VKUg"
+        }
+        os.environ['PUBLIC_KEY'] = json.dumps(self.pem)
```

Comment:
setUp is run for all middleware tests, so setting env variables here also affets non-authentication tests. I'd suggest that you isolate authentication tests to new testcase class.